cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -86,32 +168,81 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
 
   @Override
   public List<String> getScheduleIds(Id.Program program, SchedulableProgramType programType) {
-    return ImmutableList.of();
+    Set<String> scheduleIds;
+    synchronized (this) {
+      scheduleIds = ImmutableSet.copyOf(scheduleSubscribers.keySet());
+    }
+    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    String programScheduleId = getProgramScheduleId(program, programType);
+    for (String scheduleId : scheduleIds) {
+      if (scheduleId.startsWith(programScheduleId)) {
+        builder.add(scheduleId);
+      }
+    }
+    return builder.build();
   }
 
   @Override
   public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.suspendScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.resumeScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void deleteSchedule(Id.Program programId, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.remove(getScheduleId(programId, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.deleteSchedule(programId, programType, scheduleName);
+      synchronized (this) {
```

Comment:
I am not sure I understand. A subscriber can multiple tasks? You only cancel the subscriber, if this was the only task for that subscriber. But if it has other tasks, then it is already removes from the scheduleSubscribers... So there is no reference to it anymore, but it has more active tasks?